examples/profiling/primitive_instrumentation.cpp

```diff
@@ -0,0 +1,205 @@
+//   Copyright (c) 2018 Parsa Amini
+//   Copyright (c) 2018 Hartmut Kaiser
+//
+//   Distributed under the Boost Software License, Version 1.0. (See accompanying
+//   file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/phylanx.hpp>
+
+#include <hpx/hpx_init.hpp>
+#include <hpx/include/agas.hpp>
+#include <hpx/runtime_fwd.hpp>
+
+#include <cstdint>
+#include <iostream>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+///////////////////////////////////////////////////////////////////////////////
+char const* const lra_code = R"(block(
+    //
+    // Logistic regression analysis algorithm
+    //
+    //   x: [30, 2]
+    //   y: [30]
+    define(lra, x, y, alpha,
+        block(
+            define(weights, constant(0.0, shape(x, 1))),     // weights: [2]
+            define(transx, transpose(x)),                    // transx:  [2, 30]
+            define(pred, constant(0.0, shape(x, 0))),
+            define(error, constant(0.0, shape(x, 0))),
+            define(gradient, constant(0.0, shape(x, 1))),
+            define(step, 0),
+            while(
+                step < 10,
+                block(
+                    store(pred, 1.0 / (1.0 + exp(-dot(x, weights)))),
+                    store(error, pred - y),                  // error: [30]
+                    store(gradient, dot(transx, error)),     // gradient: [2]
+                    parallel_block(
+                        store(weights, weights - (alpha * gradient)),
+                        store(step, step + 1)
+                    )
+                )
+            ),
+            weights
+        )
+    ),
+    lra
+))";
+
+// Retrieve performance counter data for selected primitives
+//   primitive_instances:     The primitives for which performance counter data
+//                            is required
+//   locality_id:             The locality the performance counter data is going
+//                            to be queried from
+//   counter_name_last_parts: A vector containing the last part of the
+//                            performance counter names.
+//                            e.g. std::vector{ "count/eval", "count/direct_eval" }
+// NOTE: primitive_instances are not verified
+std::map<std::string, std::vector<std::int64_t>> retrieve_counter_data(
```

Comment:
I'd suggest to move the locality-id to the last position as this would allow to leave it with a default value 
```
    ..., hpx::id_type const& locality_id = hpx::find_here())
```

Reply:
Fixed.